### PR TITLE
CLM-14: Add Sample icon should not appear for Test Types that don't require sample in main test orders table

### PR DESCRIPTION
**Pull request by tahiraniazi**
**Issue I worked on**

https://issues.openmrs.org/browse/CLM-14

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/commonlabtest/web/controller/LabTestOrderPortletController.java [hunk 1]
```diff
@@ -67,10 +68,13 @@ protected void populateModel(HttpServletRequest request, Map<String, Object> mod
 						orderJsonArray.add(childJsonObject);
 					}
 				}
+				boolean anyTestRequireSample = false;
+				for (JsonElement element : orderJsonArray) {
+					anyTestRequireSample = ((JsonObject) element).get("requiredSpecimen").getAsBoolean();
+				}
 				System.out.println("Totatl Resullt SET ::" + orderJsonArray);
```
Comment (seekme94) [inline omod/src/main/java/org/openmrs/module/commonlabtest/web/controller/LabTestOrderPortletController.java:75]: Remove console output statements. Rather put them in the log if this information is to be collected.

File: omod/src/main/java/org/openmrs/module/commonlabtest/web/controller/LabTestOrderPortletController.java [hunk 2]
```diff
@@ -67,10 +68,13 @@ protected void populateModel(HttpServletRequest request, Map<String, Object> mod
 						orderJsonArray.add(childJsonObject);
 					}
 				}
+				boolean anyTestRequireSample = false;
+				for (JsonElement element : orderJsonArray) {
+					anyTestRequireSample = ((JsonObject) element).get("requiredSpecimen").getAsBoolean();
```
Comment (seekme94) [inline omod/src/main/java/org/openmrs/module/commonlabtest/web/controller/LabTestOrderPortletController.java:73]: Not making sense. If there are 2 elements; the first one has requiredSpecimen = true and second one has requiredSpecimen = false then anyTestRequireSample will be false when the loop ends. Are we missing a break statement, or some other operation is to be done inside the loop?

Comment (tahiraniazi) [reply omod/src/main/java/org/openmrs/module/commonlabtest/web/controller/LabTestOrderPortletController.java:73]: Fixed! probably an undo resulted in this.

File: omod/src/main/java/org/openmrs/module/commonlabtest/web/controller/LabTestOrderPortletController.java [hunk 3]
```diff
@@ -67,10 +68,14 @@ protected void populateModel(HttpServletRequest request, Map<String, Object> mod
 						orderJsonArray.add(childJsonObject);
 					}
 				}
-				System.out.println("Totatl Resullt SET ::" + orderJsonArray);
+				boolean anyTestRequireSample = false;
+				for (JsonElement element : orderJsonArray) {
+					anyTestRequireSample = ((JsonObject) element).get("requiredSpecimen").getAsBoolean();
+					if (anyTestRequireSample == true)
```
Comment (seekme94) [inline omod/src/main/java/org/openmrs/module/commonlabtest/web/controller/LabTestOrderPortletController.java:74]: Quick tip. if (anyTestRequiredSample) was sufficient.

**Review by seekme94 — APPROVED** [review]

File: omod/src/main/java/org/openmrs/module/commonlabtest/web/controller/LabTestOrderPortletController.java (hunk 3 above)
Comment (tahiraniazi) [reply omod/src/main/java/org/openmrs/module/commonlabtest/web/controller/LabTestOrderPortletController.java:74]: yes I know.. both are fine

**Pull request merged**